linera-execution/src/execution_state_actor.rs

```
@@ -425,6 +431,70 @@ where
     }
 }
 
+impl<C> ExecutionStateView<C>
+where
+    C: Context + Clone + Send + Sync + 'static,
+    C::Extra: ExecutionRuntimeContext,
+{
+    /// Receives an HTTP response, returning the prepared [`http::Response`] instance.
+    ///
+    /// Ensures that the response does not exceed the provided `size_limit`.
+    async fn receive_http_response(
+        &mut self,
+        mut response: reqwest::Response,
+        size_limit: u64,
+    ) -> Result<http::Response, ExecutionError> {
+        let status = response.status().as_u16();
+        let maybe_content_length = response.content_length();
+
+        let headers = mem::take(response.headers_mut())
+            .into_iter()
+            .scan(None, |previous_name, (maybe_name, value)| {
+                let name = maybe_name
+                    .or(previous_name.take())
+                    .expect("All headers should have names");
```

Do we catch this panic?

---

OK but we shouldn't let panic in this crate crash our nodes. I'd say it's the problem with the server that responds to the query so the execution should simply fail.

---

I'm gonna insist that we don't panic here.

---

This is not caused by a problem with the server, but the presence of a bug in the `reqwest` crate.

But let me see if I can refactor it to avoid the `expect`.

---

We can return an error here and handle that on the caller site.

---

I did something different. I think it's better now.